django/db/migrations/executor.py

```diff
@@ -224,6 +224,7 @@ def _migrate_all_backwards(self, plan, full_plan, fake):
         # remaining applied migrations.
         last_unapplied_migration = plan[-1][0]
         state = states[last_unapplied_migration]
+        del state.apps
```

Comment:
This speeds up mutation - but do we need to re-render `state.apps` after mutation is finished for the benefit of any post-migrate signal? 🤔

Reply:
yes, there's no way around complete model rendering here due to `post_migrate` eager rendering of models

I could see how disposing of rendered models like you did you help here though. What about using `state.bulk_update()` instead?

Reply:
Assuming you meant `state.apps.bulk_update()`? Unfortunately still 20s to unapply _only_ the final migration.

Reply:
Results of `del state.apps` on my project:
```
|-------------------------+--------+----------------|
|                         | Normal | del state.apps |
|-------------------------+--------+----------------|
| Unapply final migration | 20.5s  | 3.7s           |
| Unapply all migrations  | 177.8s | 175.08s        |
|-------------------------+--------+----------------|

```
FYI this project is making use of django-pghistory so there's constant removal & re-addition of triggers used to track changes on tables. This may be affecting things: not only because of db operations but it also stays in the " Rendering model states..." state for quite a while.

Reply:
Can we add a comment to explain why this is helpful?
```
# Delete apps to avoid ...
```

Reply:
Yep I have to go back and remember why that was myself 😅